### PR TITLE
Update parameter names in card component

### PR DESCRIPTION
**Pull request by precious-onyenaucheya-ons**
### What is the context of this PR?

[100](https://github.com/ONSdigital/design-team/issues/100)

As outlined in the technical notes of the ticket, the following parameters have been updated in the card component:

1. `headingLevel`, `id`, `titleClasses`, `title`, and `url` have been combined into a single object (named `title`) with `headingLevel`, `id`, `classes`, `text`, and `url` properties.
2. `text`, `textId`, and `itemsList` have been merged into a single object (named `body`) with `text`, `id`, and `itemsList` properties.

Additionally, the `_macro-options.md` file has been updated.

#### _BREAKING CHANGES_

- **Description of change:** 
  1. `headingLevel`, `id`, `titleClasses`, `title`, and `url` are now combined into a `title` object with `headingLevel`, `id`, `classes`, `text`, and `url` properties.
  2. `text`, `textId`, and `itemsList` have been consolidated into a `body` object with `text`, `id`, and `itemsList` properties.

- **Reason for change:** This update ensures consistency in parameter naming across the project.

- **Migration steps:**
  1. Locate all instances of the `headingLevel`, `id`, `titleClasses`, `title`, and `url` parameters.
  2. Restructure these parameters into the `title` object, including `headingLevel`, `id`, `classes`, `text`, and `url`.
  3. Locate all instances of the `text`, `textId`, and `itemsList` parameters.
  4. Restructure these into the `body` object, containing `text`, `id`, and `itemsList`.

<details>
        <summary><b>Click for example</b></summary>
        
        OLD
         {{
              onsCard({
                "id": 'card-example',
                "textId": 'text',
                "title": 'Your data and security',
                "url": '#0',
                "text": 'How we keep your data safe and what happens to your personal information.',
                "headingLevel": 2,
                "classes": "random-class",
                "itemsList": [
                    {
                        "url": '#0',
                        "text": 'List item 1 about the census'
                    },
                    {
                        "url": '#0',
                        "text": 'List item 2 about the census'
                    }
                ]
            })
         }}

    
          NEW
          {{
              onsCard({
                    "title": {
                        "id": 'card-example',
                        "text": 'Your data and security',
                        "url": '#0'
                        "headingLevel": 2,
                        "titleClasses": "random-class"
                      },
                      "body":{
                          "id": 'text',
                          "text": 'How we keep your data safe and what happens to your personal information.',
                          "itemsList": [
                                  {
                                      "url": '#0',
                                      "text": 'List item 1 about the census'
                                  },
                                  {
                                      "url": '#0',
                                      "text": 'List item 2 about the census'
                                  }
                              ]
                      }
              })
          }}
 </details>

 
### How to review this PR

Check that all the references to the parameters have been updated
Check that all the visual tests pass

-   [x] I have selected the correct Assignee
-   [x] I have linked the correct Issue

### CODE REVIEW

File: src/components/card/_macro-options.md [hunk 1]
```diff
@@ -18,3 +12,21 @@
 | largeSrc       | string | false    | Path to the retina version of the image                                                                              |
 | alt            | string | false    | The HTML `alt` tag to explain the appearance and function of the image. Not required if the image is only decorative |
 | placeholderURL | string | false    | Optional base path for placeholder image                                                                             |
+
+## Title
+
+| Name         | Type   | Required | Description                                                                   |
+| ------------ | ------ | -------- | ----------------------------------------------------------------------------- |
+| text         | string | true     | The text for the title of the card heading                                    |
```
Comment (rmccar) [inline src/components/card/_macro-options.md:20]: This wording is confusing maybe change the description to "The text for the card title"

File: src/components/card/_macro-options.md [hunk 2]
```diff
@@ -18,3 +12,21 @@
...
+| headingLevel | int    | false    | Number used to determine the heading level of the card title. Defaults to `2` |
+| classes      | string | false    | Font size classes for the card heading. Defaults to `ons-u-fs-m`              |
```
Comment (rmccar) [inline src/components/card/_macro-options.md:22]: Changing "heading" to "title" is more consistent with other descriptions

File: src/components/card/_macro-options.md [hunk 3]
```diff
@@ -18,3 +12,21 @@
...
+| url          | string | true     | The URL for the title link `href` attribute                                   |
+| id           | string | true     | The HTML `id` attribute for the card heading                                  |
```
Comment (rmccar) [inline src/components/card/_macro-options.md:24]: Again "title" would make more sense with the other desciptions

Comment (rmccar) [inline src/components/card/_macro-options.md:24]: Again "title" would make more sense with the other descriptions

**Review by rmccar — COMMENTED** [review]
Other than these minor things looks good

File: src/components/card/_macro-options.md (hunk 1 above)
Comment (precious-onyenaucheya-ons) [reply src/components/card/_macro-options.md]: done

File: src/components/card/_macro-options.md (hunk 2 above)
Comment (precious-onyenaucheya-ons) [reply src/components/card/_macro-options.md]: updated

File: src/components/card/_macro-options.md (hunk 3 above)
Comment (precious-onyenaucheya-ons) [reply src/components/card/_macro-options.md]: updated

**Review by rmccar — APPROVED** [review]

**Review by SriHV — APPROVED** [review]

**Pull request merged**